### ISSUE TITLE
no-restricted-properties doesn't work for properties of function calls

### ISSUE THREAD

**Issue by osdiab** [eslint/eslint#13823]
**Tell us about your environment**

* **ESLint Version:** 6.6.0
* **Node Version:** 12.19.0
* **npm Version:** 6.14.8

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
typescript-eslint/parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  rules: {
    "no-restricted-properties": [
      "error",
      {
        property: "localStorage",
        message: "Use getLocalStorage() function to access local storage",
      },
    ],
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function getWindow() {
  // eslint-disable-next-line no-restricted-globals
  return typeof window === "undefined" ? undefined : window;
}

getWindow()?.localStorage.setItem("hi", "bye")
```

```bash
eslint --ignore-path .gitignore --ext "js,ts,jsx,tsx" ./src
```

**What did you expect to happen?**
the line accessing `localStorage` from `getWindow()` would fail

**What actually happened? Please include the actual, raw output from ESLint.**
it didn't

**Are you willing to submit a pull request to fix this bug?**
yes

**Comment by mdjermanovic** [comment]
Hi @osdiab, thanks for the issue!

I can't reproduce this with the default parser, here's a [demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgXCJuby1yZXN0cmljdGVkLXByb3BlcnRpZXNcIjogW1xuICAgICAgXCJlcnJvclwiLFxuICAgICAge1xuICAgICAgICBwcm9wZXJ0eTogXCJsb2NhbFN0b3JhZ2VcIixcbiAgICAgICAgbWVzc2FnZTogXCJVc2UgZ2V0TG9jYWxTdG9yYWdlKCkgZnVuY3Rpb24gdG8gYWNjZXNzIGxvY2FsIHN0b3JhZ2VcIixcbiAgICAgIH0sXG5dKi9cblxuZ2V0V2luZG93KCk/LmxvY2FsU3RvcmFnZS5zZXRJdGVtKFwiaGlcIiwgXCJieWVcIik7XG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEyLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19).

This is most likely caused by an old version of `@typescript-eslint/parser` where optional chaining had a different AST representation.

Can you upgrade `@typescript-eslint` and ESLint to the latest versions?

**Comment by osdiab** [comment]
I'll give it a go and report back soon! Pretty likely, our typescript-eslint parser version is probably a bit dated.
